\documentclass[t]{beamer}
\usepackage{listings}
\usepackage{minted}

\usetheme{default}
\usebackgroundtemplate{\includegraphics[width=\paperwidth]
                                       {../cpeb_bkground_topleftlogo.pdf}}

\setbeamertemplate{frametitle}{
  \centering\vspace{1mm}\insertframetitle\par\vspace{3mm}
}

\usepackage[style=nature,
            hyperref,
            backend=biber,
            isbn=false,
            doi=false,
            url=false,
            date=year,
            maxbibnames=3
           ]{biblatex}

\bibliography{kdm.bib}

\title{\texttt{kWIP}: The k-mer Weighted Inner Product}
\subtitle{And the rest of my PhD\ldots}
\author{Kevin Murray}
\institute{Borevitz Lab, CPEB, ANU}
\date{\today}

\usefonttheme{serif}

\begin{document}

{
\usebackgroundtemplate{\includegraphics[width=\paperwidth]{../cpeb_bkground_centered.pdf}}
\begin{frame}
  \titlepage
  \vfill
\end{frame}
}

\begin{frame}{Large-scale population genomics}
  \begin{itemize}
    \item Moving from 100s to 1,000s or 10,000s of samples \emph{per study!}
    \item Efficient algorithms to analyse large-scale genomic data
    \begin{itemize}
      \item Reference \& alignment free: \textit{less bias, de novo}
      \item Platform/protocol agnostic: \textit{future proof}
      \item Computationally efficient: \textit{not the bottleneck}
    \end{itemize}
  \end{itemize}
  \begin{center}
    \includegraphics[width=\textwidth]{img/cross-scale.png}
  \end{center}
  \tiny{after \textcite{peterson_double_2012}}
\end{frame}

\begin{frame}{Genetic Similarity Estimation}
  \begin{itemize}
    \item Rough approximation of sample relatedness required
      \begin{itemize}
        \item For natural collections
        \item As a technical control
      \end{itemize}
  \end{itemize}
  \begin{center}
    \includegraphics<1>[width=\textwidth]{img/restruct-1}
    \includegraphics<2>[width=\textwidth]{img/restruct-2}
    \includegraphics<3>[width=\textwidth]{img/restruct-3}
    \begin{itemize}
      \item[]<1-3> \tiny{after \textcite{brachi_genome-wide_2011}}
    \end{itemize}
    \includegraphics<4>[width=\textwidth]{img/jared-tree.pdf}
    \includegraphics<5>[width=0.6\textwidth]{img/at80-tree.png}
  \end{center}
\end{frame}


\begin{frame}[c]{Genetic Similarity Estimation}
  \centering Initially, we care mostly about the deepest and shallowest
  branches of the tree.
\end{frame}

\begin{frame}{Presenting \texttt{kWIP}}
  \begin{itemize}
    \item $k$-mer based \textit{de novo} genetic relatedness estimator
    \item Weighted Inner Product between hashes to determine relatedness
    \item Produces a distance matrix from raw NGS reads
  \end{itemize}
  \begin{center}
    \includegraphics[width=\textwidth]{img/kwip-overview.png}
  \end{center}
\end{frame}

\begin{frame}{Technological Overview}
  \begin{itemize}
    \item Alignment-free Sequence Comparison
    \item Hashing and Probabilistic Data Structures
    \item Population and Frequency Hashes
    \item (Weighted) Inner Products
  \end{itemize}
\end{frame}

% \begin{frame}{$k$-mer Seq. Comparison}
%   \begin{columns}[t]
%     \column{0.5\textwidth}
%       \begin{itemize}
%         \item<1-> Analyse sequences by decomposition to $k$-length words
%         \item<1-> Computationally and biologically appropriate
%         \begin{itemize}
%           \item<1-> Fast: Scalable and parallelisable
%           \item<1-> Constant-memory (using \texttt{khmer})
%           \item<1-> Reference-free
%           \item<1-> Easily scales to longer or more reads \end{itemize} \end{itemize}
%     \column{0.5\textwidth}
%     \begin{center}
%       \includegraphics<1>[height=0.6\textheight]{img/ali.png}
%       \includegraphics<2>[height=0.6\textheight]{img/ali-free.png}
%     \end{center}
%   \end{columns}
% \end{frame}

\begin{frame}{$k$-mer Sequence Comparison}
  \begin{itemize}
    \item Many existing tools
    \begin{itemize}
      \item Feature Frequency Profiles \autocite{sims_alignment-free_2009}
      \item $D2$ and related statistics
      \item \texttt{spaced} and other spaced-word approaches
        \autocite{morgenstern_estimating_2015,leimeister_fast_2014}
      \item \texttt{Cnidaria} and other Jaccard distance approaches
        \autocite{aflitos_cnidaria:_2015}
      \item Early steps in BLAST \textit{et al.}
    \end{itemize}
    \item Most require assembled gene/genome sequence
    \item Many use inner product ($D2$) as similarity measure
  \end{itemize}
\end{frame}


\begin{frame}{Entropy Weighting}
  \begin{itemize}
    \item \texttt{kWIP} weights by Shannon entropy: $H(frequency)$
    \item Shannon entropy: measure of information
    \item $-(P_i log_2(P_i) + (1-P_i) log_2(1-P_i))$
  \end{itemize}
  \begin{center}
    \includegraphics[width=0.6\textwidth]{img/shanent.png}
  \end{center}
\end{frame}

\begin{frame}{\texttt{kWIP} Algorithm}
  \begin{itemize}
    \item For each run: count all $k$-mers into a hash
    \item For each analysis set:
      \begin{itemize}
        \item Calculate the entropy of $k$-mer frequency ($H$)
        \item For each pair of runs with $k$-mer counts $A$ and $B$, calculate
          $$\sum\limits^{n}_{i=1} A_i \cdot B_i \cdot H_i$$
      \end{itemize}
  \end{itemize}
  \begin{center}
    \includegraphics[width=0.4\textwidth]{img/hash-wip.png}
  \end{center}
\end{frame}


\begin{frame}{\texttt{kWIP}}
  \begin{itemize}
    \item The software:
      \begin{itemize}
        \item \texttt{C++}, $>$2000 lines of code
        \item Uses \texttt{khmer} for $k$-mer counting \& hashing
        \item Parallelised, fast
        \item GNU GPL licensed, source code on GitHub
      \end{itemize}
      \begin{center}
        \includegraphics[width=0.5\textwidth]{img/kwip-doc-screenshot.png}
      \end{center}
  \end{itemize}
\end{frame}

\begin{frame}{\texttt{kWIP} Case Studies}
  \begin{itemize}
    \item 3000 rice genomes
      \begin{itemize}
        \item 3000 rice samples (25k runs)
        \item Analysing in sets of 96, from all major groups
        \item \textcite{the_3000_rice_genomes_project_3000_2014}
      \end{itemize}
    \item Chlamydomonas
      \begin{itemize}
        \item $\approx 20$ lines from USA
        \item High coverage re-sequencing with leftover assembly
        \item Compare to SNP-based distance calculation
        \item \textcite{flowers_whole-genome_2015}
      \end{itemize}
    \item Simulation
      \begin{itemize}
        \item Fake population genome sequencing studies
      \end{itemize}
  \end{itemize}
\end{frame}

\begin{frame}{96 Rice Runs}
  \begin{itemize}
    \item Set of 96 rice runs from 16 samples (6 tech reps ea)
    \item About half/half from 2 major groups (Indica, Japonica)
    \item Expectations:
      \begin{itemize}
        \item All runs cluster into groups of 6 reps (16 samples)
        \item Big split between two groups: (7 and 9 respectively here)
      \end{itemize}
    \item Recover known grouping w/ \texttt{kWIP}, not w/ unweighted IP
    \item Sensitive to read depth
    \item Took $\approx 6$ hours on 16 CPU, 64GB RAM supercomputer node
  \end{itemize}
\end{frame}

\begin{frame}
  \begin{center}
    \includegraphics<1>[width=0.6\textwidth]{img/dendro-wip.png}
    \includegraphics<2>[width=0.6\textwidth]{img/dendro-ip.png}
  \end{center}
\end{frame}

\begin{frame}{Chlamydomonas}
  \begin{center}
    \includegraphics<1>[width=0.6\textwidth]{img/chlamydomonas_PCA_from_paper.png}
    \includegraphics<2>[width=0.6\textwidth]{img/chlamydomonas_PCA_full-set-dim_1-3.png}
    \begin{itemize}
      \item Population genomics of Chlamy in USA
      \item[] \tiny{Data from \textcite{flowers_whole-genome_2015}}
    \end{itemize}
  \end{center}
\end{frame}

\begin{frame}{Simulation}
  \begin{itemize}
    \item Perform simulated sequencing experiment:
    \begin{itemize}
      \item Simuate natural population structure
      \item Simulate sample genomes, sequencing runs
      \item Hash reads, kWIP
    \end{itemize}
    \item kWIP quantitatively outperforms unweighted equivalent
      \begin{itemize}
        \item Can compare known truth to kWIP results
        \item Spearmans Rank Correlation ($\rho$)
      \end{itemize}
  \end{itemize}
\end{frame}

\begin{frame}{Simulation Results}
  \begin{figure}
    \centering
    \includegraphics[width=0.6\textwidth]{img/cov_vs_rho_1000.pdf}
    \caption{Coverage vs Accuracy}
  \end{figure}
\end{frame}

\begin{frame}{Simulation Results}
  \begin{figure}
    \centering
    \includegraphics[width=0.6\textwidth]{img/variation_vs_accuracy.pdf}
    \caption{Average variation vs Accuracy}
  \end{figure}
\end{frame}


\begin{frame}{Conclusions}
  \begin{itemize}
    \item \texttt{kWIP} is implemented, Beta software
    \item Publically available at \url{github.com/kdmurray91/kwip}
    \item We show the utility of \texttt{kWIP}
    \item Publication coming soon
    \pause
    \item What's next?
      \begin{itemize}
        \item Norman will lead the kWIP project's evolution
        \item I will focus on assembly
      \end{itemize}
  \end{itemize}
\end{frame}

\begin{frame}{Thanks}
  \begin{itemize}
    \item My collaborators
      \begin{itemize}
        \item Cheng Soon Ong
        \item Christfried Webers
        \item Norman Warthmann
        \item Sylvain For\^{e}t
      \end{itemize}
    \item Supervisors:
      \begin{itemize}
        \item Justin Borevitz
        \item Gavin Huttley
        \item Barry Pogson
      \end{itemize}
    \item \texttt{khmer} folks (DIB-lab, UC Davis)
      \begin{itemize}
        \item C. Titus Brown
        \item Michael Crusoe
        \item Camille Scott
      \end{itemize}
    \item Yourselves
  \end{itemize}
\end{frame}

\begin{frame}[shrink=20]{}
  \printbibliography
  \vfill
  .
\end{frame}

\end{document}
