\documentclass[t]{beamer}
\usepackage{listings}
\usepackage{minted}

\usetheme{default}
\usebackgroundtemplate{\includegraphics[width=\paperwidth]
                                       {../cpeb_bkground_topleftlogo.pdf}}

\setbeamertemplate{frametitle}{
  \centering\vspace{1mm}\insertframetitle\par\vspace{3mm}
}

\usepackage[style=nature,
            hyperref,
            backend=biber,
            isbn=false,
            doi=false,
            url=false,
            date=year,
            maxbibnames=3
           ]{biblatex}

\bibliography{kwip-tic.bib}

\title{\texttt{kWIP}: The k-mer Weighted Inner Product}
\author{Kevin Murray}
\institute{PhD Candidate, Borevitz Lab, CPEB, ANU}
\date{21 August 2015}

\usefonttheme{serif}

\begin{document}

{
\usebackgroundtemplate{\includegraphics[width=\paperwidth]{../cpeb_bkground_centered.pdf}}
\begin{frame}
  \titlepage
  \vfill
\end{frame}
}

\begin{frame}{Disclaimer}
  \begin{center}
    \vfill
    \large{\texttt{DRAFT}}
    \\
    \pause
    \vfill
    \small{Or as Norman says, \texttt{kWIP} is Kevin's Work In Progress}
    \vfill
  \end{center}
\end{frame}


\begin{frame}{Collaboration}
  \begin{itemize}
    \item This work is a collaborative effort
      \begin{itemize}
        \item Norman Warthmann
        \item Cheng Soon Ong
        \item Chris Webers
      \end{itemize}
  \end{itemize}
\end{frame}

\begin{frame}{Overview}
  \begin{itemize}
    \item Motivation
    \item Technological overview
    \item Early results and plans
    \item Demonstration
  \end{itemize}
\end{frame}

\begin{frame}{Large-scale population genomics}
  \begin{itemize}
    \item Moving from 100s to 1,000s or 10,000s of samples \textit{per PhD!}
      \pause
    \item Efficient algorithms to analyse large-scale genomic data
    \begin{itemize}
      \item Reference \& alignment free: \textit{less bias, de novo}
      \item Platform/protocol agnostic: \textit{future proof}
      \item Computationally efficient: \textit{not the bottleneck}
      \item Cross scale: \textit{one tool to rule them all}
    \end{itemize}
  \end{itemize}
  \begin{center}
    \includegraphics[width=\textwidth]{img/cross-scale.png}
  \end{center}
  \tiny{after \textcite{peterson_double_2012}}
\end{frame}


\begin{frame}{Rapid and Basic Clustering}
  \begin{itemize}
    \item Rough approximation of sample relatedness required
      \begin{itemize}
        \item For natural collections
        \item For association mapping
        \item As a technical control
      \end{itemize}
  \end{itemize}
  \pause
  \begin{center}
    \includegraphics<2>[width=\textwidth]{img/restruct-1}
    \includegraphics<3>[width=\textwidth]{img/restruct-2}
    \includegraphics<4>[width=\textwidth]{img/restruct-3}
    \includegraphics<5>[width=\textwidth]{img/restruct-4}
  \end{center}
  \tiny{after \textcite{brachi_genome-wide_2011}}
\end{frame}

\begin{frame}[fragile]{\textit{De novo} technical control}
  \begin{itemize}
    \item Sample DNA not very physically distictive
      \begin{itemize}
        \item Mix-ups and contamination occur
      \end{itemize}
    \item Not just for your own data: SRA not so perfect
  \end{itemize}
  \begin{center}
    \includegraphics[width=0.6\textwidth]{img/at80-tree.png}
  \end{center}
  \pause
  \begingroup
    \fontsize{7pt}{7pt}\selectfont
    \begin{verbatim}
      7d74f1b53174afc9fd3aec1dde1d996e SRR095715.fastq SRR095818.fastq
      54da64f0343b69bcb959d33f127505d8 SRR095713.fastq SRR095817.fastq
    \end{verbatim}
  \endgroup
\end{frame}

\begin{frame}{Presenting \texttt{kWIP}}
  \begin{itemize}
    \item $k$-mer based \textit{de novo} genetic clustering
    \item Weighted Inner Product between hashes to determine similarity
    \item Produces a distance matrix from raw NGS reads
  \end{itemize}
  \begin{center}
    \includegraphics[width=\textwidth]{img/kwip-overview.png}
  \end{center}
\end{frame}

\begin{frame}{Technological Overview}
  \begin{itemize}
    \item $k$-mer analysis
    \item Hashing and Probablistic Data Structures
    \item Population and Frequency Hashes
    \item (Weighted) Inner Products
  \end{itemize}
\end{frame}

\begin{frame}{$k$-mer analysis}
  \begin{columns}[t]
    \column{0.6\textwidth}
      \begin{itemize}
        \item<1-> Analyse $k$-length words of sequences
        \item<2> Computationally and biologically appropriate
        \begin{itemize}
          \item<2> Alignment Free
          \item<2> Constant-memory (using \texttt{khmer})
          \item<2> Fast: Scalable and parallelisable
          \item<2> Cross-scale
        \end{itemize}
      \end{itemize}
      \vfill
    \column{0.4\textwidth}
    $k = 3$\\
    \texttt{ACGTGT}\\
    \texttt{ACG~~~}\\
    \texttt{~CGT~~}\\
    \texttt{~~GTG~}\\
    \texttt{~~~TGT}
  \end{columns}
\end{frame}

\begin{frame}{Hashes and Hash Functions}
  \begin{itemize}
    \item Hash function e.g. \mint{python}|hash('ACG') => 5234315134|
    \item For DNA, 2-bit encoding is used
      \pause
    \item ``Hash'': a probablistic data structure
      \begin{itemize}
        \item Efficent way of counting k-mers
        \item Constant memory
        \item Easy set operations and inner product
        \item Implicit de Bruijn graph
        \item Implemented in C Titus Brown's \texttt{khmer}
      \end{itemize}
  \end{itemize}
\end{frame}

\begin{frame}{Hash}
  \begin{itemize}
    \item Subset of a Count-Min Sketch/Counting Bloom Filter
    \item Vector of large prime length (e.g. 1e9 + 7)
    \item Indexed modulo length $(bin = h(w_i) \mod prime)$
    \item Aliasing can occur, hence probablistic
  \end{itemize}
  \begin{center}
    \includegraphics<1>[width=0.6\textwidth]{img/hash-0.png}
    \includegraphics<2>[width=0.6\textwidth]{img/hash-1.png}
    \includegraphics<3>[width=0.6\textwidth]{img/hash-2.png}
    \includegraphics<4>[width=0.6\textwidth]{img/hash-3.png}
    \includegraphics<5>[width=0.6\textwidth]{img/hash-4.png}
  \end{center}
\end{frame}


\begin{frame}{Hash Operations}
  \begin{itemize}
    \item Population sum, or hash frequency
  \end{itemize}
  \begin{center}
    \includegraphics<1>[width=0.6\textwidth]{img/hash-sums.png}
    \includegraphics<2>[width=0.6\textwidth]{img/hash-sumfreq.png}
  \end{center}
\end{frame}

\begin{frame}{Shannon Entropy}
  \begin{itemize}
    \item Measure of Information
    \item $ -\sum\limits_{i} p_i log(p_i) $
    \item \texttt{kWIP} weights by $H(frequency)$
  \end{itemize}
  \begin{center}
    \includegraphics[width=0.6\textwidth]{img/shanent.pdf}
  \end{center}
\end{frame}


\begin{frame}{$k$-mer based clustering}
  \begin{itemize}
    \item Alignment-free sequence clustering is a whole field
    \item $D2$ and friends
    \item Most require assembled gene/genome sequence
    \item Many use inner product as similarity measure
  \end{itemize}
  \begin{center}
    \includegraphics[width=0.6\textwidth]{img/foret-et-al-d2.png}
  \end{center}
\end{frame}

\begin{frame}{\texttt{kWIP}}
  \begin{itemize}
    \item The $k$-mer Weighted Inner Product
      \begin{itemize}
        \item Extends alignment-free seq comparison to raw NGS data
      \end{itemize}
    \pause
    \item Algorithm:
      \begin{itemize}
        \item For each sample: count all $k$-mers into a hash
        \pause
        \item For each analysis set, i.e ``population'':
          \begin{itemize}
            \item Calculate the entropy of population frequency ($H$)
            \item For each pair of samples $A$ and $B$, calculate \\
              $\sum\limits^{n}_{i=0} A_i \cdot B_i \cdot H_i$
          \end{itemize}
      \end{itemize}
  \end{itemize}
  \begin{center}
    \includegraphics[width=0.4\textwidth]{img/hash-wip.png}
  \end{center}
\end{frame}


\begin{frame}{\texttt{kWIP}}
  \begin{itemize}
    \item The software:
      \begin{itemize}
        \item \texttt{C++}, $>$2000 lines of code
        \item Uses \texttt{khmer} for $k$-mer counting \& hashing
        \item Parallelised, fast
        \item GNU GPL licensed, source code on GitHub
      \end{itemize}
      \begin{center}
        \includegraphics[width=0.5\textwidth]{img/kwip-doc-screenshot.png}
      \end{center}
  \end{itemize}
\end{frame}

\begin{frame}{\texttt{kWIP} Experiments}
  \begin{itemize}
    \item 3000 rice genomes:
      \begin{itemize}
        \item 3000 rice lines from known families
        \item Analysing in sets of $\approx 100$, from all major groups
        \item Recover known grouping w/ \texttt{kWIP}, not w/ unweighted IP
        \item Sensitive to read depth
      \end{itemize}
    \item Simulation
      \begin{itemize}
        \item Fake population genome sequencing studies
      \end{itemize}
  \end{itemize}
\end{frame}

\begin{frame}{96 Rice Runs}
  \begin{itemize}
    \item Set of 96 rice runs from 16 samples (6 tech reps ea)
    \item About half/half from 2 major groups (Indica, Japonica)
    \item Expectations:
      \begin{itemize}
        \item All runs cluster into groups of 6 reps (16 samples)
        \item Big split between two groups: (7 and 9 respectively here)
      \end{itemize}
    \item We see this with \texttt{kWIP}, not with Unweighted IP
    \item Took 10 hours on 16-core Raijin node, 60-80GB RAM
  \end{itemize}
\end{frame}

\begin{frame}
  \begin{center}
    \includegraphics<1>[width=\textwidth]{img/distmat-both.png}
    \includegraphics<2>[width=\textwidth]{img/dendro-both.png}
  \end{center}
\end{frame}

\begin{frame}
  \begin{center}
    \includegraphics<1>[width=0.6\textwidth]{img/dendro-wip.png}
    \includegraphics<2>[width=0.6\textwidth]{img/dendro-ip.png}
  \end{center}
\end{frame}

\begin{frame}{Simulation}
  \begin{itemize}
    \item Aims to re-create a biforcating tree
    \item \texttt{kWIP} does so
    \item Neighbor joining trees have Robinson-Foulds distance of 0
  \end{itemize}
  \begin{center}
    \includegraphics[width=0.6\textwidth]{img/kwip-dendro.png}
  \end{center}
\end{frame}

\begin{frame}
  \begin{center}
    \includegraphics<1>[width=0.6\textwidth]{img/true-mat.png}
    \includegraphics<2>[width=0.6\textwidth]{img/kwip-mat.png}
    \includegraphics<3>[width=0.6\textwidth]{img/ip-mat.png}
  \end{center}
\end{frame}

\begin{frame}{Simulation}
  \begin{itemize}
    \item Now for an Jupyter notebook
  \end{itemize}
\end{frame}

\begin{frame}{Thanks}
  \begin{itemize}
    \item My collaborators: Cheng Soon Ong, Christfried Webers, Norman Warthmann
    \item \{Super,Ad\}visors: Justin, Sylvain, Gavin and Barry
    \item \texttt{khmer} folks: C. Titus Brown, Michael Crusoe, Camille Scott (DIB-lab) @ UC Davis
    \item Yourselves
  \end{itemize}
\end{frame}

\begin{frame}[shrink=20]{References}
  \printbibliography
  \vfill
  .
\end{frame}


\begin{frame}{Missing heritability in the field?}
  \begin{itemize}
    \item Find more diversity in the field!
    \item Sample natural populations
      \begin{itemize}
        \item Ecological hypotheses of trait selection, adaptation
        \item Sample widely as possible across non-uniform genetic diversity
      \end{itemize}
      \pause
    \item Now \textbf{complexity limited}: complex kinship \& population
      structure
    \item Mandates development of economic, accurate large scale population
      genomics
  \end{itemize}
  \begin{center}
    \includegraphics[width=\textwidth]{img/jared-tree.pdf}
  \end{center}
\end{frame}

\end{document}
